tools/oio-event-benchmark/event_benchmark.c

```diff
@@ -0,0 +1,353 @@
+	oio_url_set(rdir_rawx_url, OIOURL_USER, RAWX_ADDRESS);
+	oio_url_set(rdir_rawx_url, OIOURL_TYPE, NAME_SRVTYPE_RDIR);
+
+	const char * const values[] = {
+		"host", FAKE_SERVICE_ADDRESS,
+		"args", "",
+		"type", NAME_SRVTYPE_RDIR,
+		"id", NAMESPACE "|" NAME_SRVTYPE_RDIR "|" FAKE_SERVICE_ADDRESS,
+		NULL
+	};
+
+	GError *err = oio_directory__force(dir, rdir_rawx_url, NAME_SRVTYPE_RDIR,
+			values);
+	if (err) {
+		GRID_ERROR("Failed to call 'reference/force': (%d) %s", err->code,
+				err->message);
+		g_clear_error(&err);
+
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static void
+free_service(struct oio_cs_registration_s *service) {
+	g_free((void *) service->id);
+	g_free((void *) service->url);
+	g_strfreev((char **) service->kv_tags);
+	g_free(service);
+}
+
+static struct oio_cs_registration_s *
+get_account_service(void)
+{
+	GSList *services = NULL;
+	void _on_reg(const struct oio_cs_registration_s *reg, int score) {
+		(void) score;
+
+		struct oio_cs_registration_s *reg_cpy =
+				g_malloc0(sizeof(struct oio_cs_registration_s));
+
+		reg_cpy->id = g_strdup(reg->id);
+		reg_cpy->url = g_strdup(reg->url);
+
+		if (reg->kv_tags) {
+			reg_cpy->kv_tags = (const char * const *)
+					g_strdupv((char **) reg->kv_tags);
+		}
+
+		services = g_slist_prepend(services, reg_cpy);
+	}
+	GError *err = oio_cs_client__list_services(cs, NAME_SRVTYPE_ACCOUNT, FALSE,
+			_on_reg);
+	if (err) {
+		GRID_ERROR("Failed to list account services: (%d) %s", err->code,
+				err->message);
+		g_clear_error(&err);
+
+		return NULL;
+	}
+
+	struct oio_cs_registration_s * account_reg = services->data;
+	g_slist_free(services);
+
+	return account_reg;
+}
+
+static gboolean
+add_fake_account(void)
+{
+	const char * const kv[6] = {
+		"tag.up", "True",
+		"tag.slots", NAME_SRVTYPE_ACCOUNT
+	};
+	struct oio_cs_registration_s reg = {
+		.id = FAKE_SERVICE_ADDRESS,
+		.url = FAKE_SERVICE_ADDRESS,
+		.kv_tags = kv
+	};
+
+	GError *err = oio_cs_client__lock_service(cs, NAME_SRVTYPE_ACCOUNT, &reg,
+			SCORE_MAX);
+	if (err) {
+		GRID_ERROR("Failed to lock service: %d %s", err->code,
+				err->message);
+		g_clear_error(&err);
+
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gpointer
+_fake_service_run(gpointer p)
+{
+	(void) p;
+
+	if (!fake_service_run()) {
+		grid_main_set_status(EXIT_FAILURE);
+		grid_main_specific_stop();
+
+		return NULL;
+	}
+
+	return NULL;
+}
+
+static gboolean
+kill_event_agent(void)
+{
+	GError *err = NULL;
+
+	gboolean success = g_spawn_command_line_sync(
+			"/usr/bin/killall oio-event-agent", NULL, NULL, NULL, &err);
+	if (err) {
+		GRID_ERROR("Command line failure': (%d) %s", err->code,
+				err->message);
+		g_clear_error(&err);
+	}
+
+	return success;
+}
+
+// Main callbacks
+
+static struct grid_main_option_s *
+grid_main_get_options(void)
+{
+	static struct grid_main_option_s cli_options[] = {
+		{
+			"Rounds", OT_UINT, {.i = &rounds},
+			"Number of rounds for a test"
+		},
```

Comment:
`Rounds` indicates the number of experiments to average. But if you want to do the same test indefinitely, you can add `Increment=0` in the options.